### PR TITLE
workspace(nix): initial docker image

### PR DESCRIPTION
**Pull request by ghuntley**
Hola.

I've never worked with circleci, so please double check I haven't stuffed something up. The docker image has been confirmed to build via `docker build -f Dockerfile .`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -217,6 +217,19 @@ jobs:
             ./.circleci/build_image.sh gecko/Dockerfile gitpod/workspace-gecko
           no_output_timeout: 60m
 
+  workspace-nix:
```
Comment (jankeromnes) [inline .circleci/config.yml:220]: Please also register this new job in the list at the bottom of this file (in order to make CircleCI actually run it). 🙏

**Review by jankeromnes — CHANGES_REQUESTED** [review]
Hey, super cool! Many thanks for this. 🙏

Please simply hook up the job (see below) so that CircleCI can build it. Then we'll know if the build passes or not, and we'll be able to test the resulting Docker image.

**Comment by shism2** [comment]
Just attaching as a reference if anyone finds it useful

`FROM buildpack-deps:focal-curl

COPY  install-packages /usr/bin

ARG DEBIAN_FRONTEND=noninteractive
ARG DUSER=gitpod
ENV HOME=/home/$DUSER

RUN install-packages \
        bash-completion \
        sudo \
        xz-utils \
        git \
# '-l': see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user
&&  useradd -l -u 33333 -G sudo -md $HOME -s /bin/bash -p $DUSER $DUSER \
# passwordless sudo for users in the 'sudo' group
&& echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers && mkdir -m 0755 /etc/nix &&  chown $DUSER /etc/nix

CMD /bin/bash -l
USER $DUSER
ENV USER $DUSER
WORKDIR $HOME

RUN  touch .bash_profile && curl https://nixos.org/releases/nix/nix-2.3.15/install | sh \
&& echo '. $HOME/.nix-profile/etc/profile.d/nix.sh' >> $HOME/.bashrc \
&& mkdir -p  ~/.config/nix/ &&  printf "sandbox = false\nexperimental-features = nix-command flakes\nkeep-derivations = true\nkeep-outputs = true" > ~/.config/nix/nix.conf \
&& mkdir -p  $HOME/.config/nixpkgs && echo '{ allowUnfree = true; }' >> $HOME/.config/nixpkgs/config.nix

RUN . $HOME/.nix-profile/etc/profile.d/nix.sh \
&& nix-channel --add https://nixos.org/channels/nixpkgs-unstable  \
&& nix-channel --update \
&& nix-channel --add https://github.com/nix-community/home-manager/archive/master.tar.gz home-manager \
&& nix-channel --update \
&& export NIX_PATH=$HOME/.nix-defexpr/channels${NIX_PATH:+:}$NIX_PATH \
&& echo 'export NIX_PATH=$HOME/.nix-defexpr/channels${NIX_PATH:+:}$NIX_PATH' >> ~/.bashrc \
&& nix-env -iA nixpkgs.nixUnstable \
&& sudo apt-get clean && sudo rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
`

**Comment by ghuntley** [comment]
@shism2 see also https://www.github.com/gitpod-io/template-nix :)

**Comment by shism2** [comment]
@ghuntley Reason I attached was to show how to skip using workspace-base

**Comment by ghuntley** [comment]
Adding in the main image instead #558

**Pull request closed without merging**